### PR TITLE
Feature/jra 134 android map editor changes

### PR DESCRIPTION
**Pull request by YawarValeriano**
Added route editor view
Added functionality for adding new points and stops
Added functionality for converting points to stops
Added functionality for deleting points
Fixed pre-commit hook on tests
Added unit tests

### CODE REVIEW

File: app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt [hunk 1]
```diff
@@ -123,25 +231,47 @@ class RouteEditorFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerCl
...
 
-    private fun addMarker(point: LatLng, withBitmap: Bitmap): Marker? {
-        return map.addMarker(MarkerOptions().position(point).icon(BitmapDescriptorFactory.fromBitmap(withBitmap)))
+    private fun addMarker(point: LatLng, withBitmap: Bitmap, isMarkerStop: Boolean): Marker? {
+        val mark = map.addMarker(MarkerOptions().position(point).icon(BitmapDescriptorFactory.fromBitmap(withBitmap)))
+        mark?.tag = isMarkerStop
+        return mark
     }
 
     override fun onMarkerClick(p0: Marker): Boolean {
```
Comment (Rodrigo-Schar) [inline app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt]: maybe you can rename this variable to something more descriptive

File: app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt [hunk 2]
```diff
@@ -123,25 +231,47 @@ class RouteEditorFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerCl
...
+            }
         }
-        changeStopButton(locationIsStop != null)
-        binding.btnAddPoint.setImageResource(R.drawable.ic_baseline_close_24)
+        binding.btnSort.isEnabled = true
+        changeRoutePointButton(false)
         return true
     }
 
     override fun onMapClick(p0: LatLng) {
```
Comment (Rodrigo-Schar) [inline app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt]: same here, but then everything looks good

**Review by Rodrigo-Schar — APPROVED** [review]

File: app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt (hunk 1 above)
Comment (YawarValeriano) [reply app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt]: Renamed

File: app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt (hunk 2 above)
Comment (YawarValeriano) [reply app/src/main/java/com/jalasoft/routesapp/ui/adminPages/routeEditor/view/RouteEditorFragment.kt]: Renamed

**Review by jmacboy — APPROVED** [review]

**Pull request merged**